template/knowledge_panels.js

```diff
@@ -0,0 +1,79 @@
+    + "</h3>\r\n"
+    + ((stack1 = lookupProperty(helpers,"each").call(depth0 != null ? depth0 : (container.nullContext || {}),(depth0 != null ? lookupProperty(depth0,"elements") : depth0),{"name":"each","hash":{},"fn":container.program(5, data, 0),"inverse":container.noop,"data":data,"loc":{"start":{"line":7,"column":0},"end":{"line":12,"column":9}}})) != null ? stack1 : "");
+},"5":function(container,depth0,helpers,partials,data) {
+    var stack1, lookupProperty = container.lookupProperty || function(parent, propertyName) {
+        if (Object.prototype.hasOwnProperty.call(parent, propertyName)) {
+          return parent[propertyName];
+        }
+        return undefined
+    };
+
+  return "  "
+    + ((stack1 = container.lambda(((stack1 = (depth0 != null ? lookupProperty(depth0,"text_element") : depth0)) != null ? lookupProperty(stack1,"html") : stack1), depth0)) != null ? stack1 : "")
+    + "\r\n"
+    + ((stack1 = lookupProperty(helpers,"if").call(depth0 != null ? depth0 : (container.nullContext || {}),((stack1 = (depth0 != null ? lookupProperty(depth0,"text_element") : depth0)) != null ? lookupProperty(stack1,"source_url") : stack1),{"name":"if","hash":{},"fn":container.program(6, data, 0),"inverse":container.noop,"data":data,"loc":{"start":{"line":9,"column":2},"end":{"line":11,"column":9}}})) != null ? stack1 : "");
+},"6":function(container,depth0,helpers,partials,data) {
+    var stack1, alias1=container.lambda, alias2=container.escapeExpression, lookupProperty = container.lookupProperty || function(parent, propertyName) {
+        if (Object.prototype.hasOwnProperty.call(parent, propertyName)) {
+          return parent[propertyName];
+        }
+        return undefined
+    };
+
+  return "  <em>source:<a href=\""
+    + alias2(alias1(((stack1 = (depth0 != null ? lookupProperty(depth0,"text_element") : depth0)) != null ? lookupProperty(stack1,"source_url") : stack1), depth0))
+    + "\">"
+    + alias2(alias1(((stack1 = (depth0 != null ? lookupProperty(depth0,"text_element") : depth0)) != null ? lookupProperty(stack1,"source_text") : stack1), depth0))
+    + "</a></em>\r\n";
+},"compiler":[8,">= 4.3.0"],"main":function(container,depth0,helpers,partials,data) {
+    var stack1, lookupProperty = container.lookupProperty || function(parent, propertyName) {
+        if (Object.prototype.hasOwnProperty.call(parent, propertyName)) {
+          return parent[propertyName];
+        }
+        return undefined
+    };
+
+  return ((stack1 = lookupProperty(helpers,"if").call(depth0 != null ? depth0 : (container.nullContext || {}),(depth0 != null ? lookupProperty(depth0,"knowledge_panels") : depth0),{"name":"if","hash":{},"fn":container.program(1, data, 0),"inverse":container.noop,"data":data,"loc":{"start":{"line":1,"column":0},"end":{"line":17,"column":7}}})) != null ? stack1 : "");
+},"useData":true});
+})();
```

Comment:
What is this file ? Is that the compilation of the handlebars template ? Why would we need to have this ?

Reply:
Yes, Its Precompiling template made by the handlebar itself for "knowledge_panels.handlebars"

Reply:
https://handlebarsjs.com/installation/precompilation.html#getting-started

Reply:
we can do something like this too, then we don't need Precompiling template
![image](https://user-images.githubusercontent.com/96618001/199959549-275de08e-c7b3-4674-a055-be0c23af65ae.png)

Reply:
Yes, I would avoid this pre-compiling. It's just a small tool for now.